test/async_scope/test_empty.cpp

```diff
@@ -161,16 +152,20 @@ TEST_CASE("TODO: async_scope is empty after adding work when in cancelled state"
   // cancel & add work
   scope.request_stop();
   bool work_executed{false};
-  scope.spawn(ex::on(sch, ex::just() | ex::then([&] { work_executed = true; })));
+  scope.spawn(ex::on(sch, ex::just() | ex::then([&] { work_executed = true;printf(".\n");})));
   // note that we don't tell impulse sender to start the work
 
-  // TODO: Make this work
-  // bool is_empty2{false};
-  // ex::sender auto snd2 = ex::on(inline_scheduler{}, scope.empty()) //
-  //                        | ex::then([&] { is_empty2 = true; });
-  // auto op2 = ex::connect(std::move(snd2), expect_void_receiver{});
-  // ex::start(op2);
-  // REQUIRE(is_empty2);
-  // REQUIRE_FALSE(work_executed);
+  bool is_empty2{false};
+  ex::sender auto snd2 = ex::on(inline_scheduler{}, scope.on_empty()) //
+                         | ex::then([&] { is_empty2 = true; });
+  auto op2 = ex::connect(std::move(snd2), expect_void_receiver{});
+  ex::start(op2);
+  REQUIRE_FALSE(is_empty2);
+
+  REQUIRE_FALSE(work_executed);
+  sch.start_next();
+  // TODO: why does this fail?
+  // REQUIRE(work_executed);
```

Comment:
Seems this should be true. Why does it fail?

Reply:
@lucteo wrote these and disabled them - I would like to have him take a look at these tests

Reply:
Ah, this is because `request_stop` has already been called on the `scope`, so the pulse scheduler completes the work with a stopped signal instead of a value. If I replace the `then` with `upon_stopped` the test passes.